### PR TITLE
Task completed GP-118: Pooling

### PR DESCRIPTION
**Pull request by Kjusok**
https://lighteatersstudio.atlassian.net/jira/software/projects/GP/boards/1?selectedIssue=GP-118

Implemented a pool for destroyed projectiles

### CODE REVIEW

File: Assets/Scripts/Gameplay/Projectiles/ProjectilePool.cs [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+        private List<Projectile> _projectiles;
+
+        [Inject]
+        public void Construct(Projectile.Factory projectileFactory)
+        {
+            _projectileFactory = projectileFactory;
+            _projectiles = new List<Projectile>();
+        }
+
+        public void AddProjectile(FlyInfo flyInfo, AttackInfo attackInfo)
```
Comment (ega-ega) [inline Assets/Scripts/Gameplay/Projectiles/ProjectilePool.cs]: Rename to "Spawn"

File: Assets/Scripts/Gameplay/Projectiles/ProjectilePool.cs [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+            projectile.Launch();
+
+            _projectiles.Add(projectile);
+        }
+
+        public void RemoveProjectile(Projectile projectile)
+        {
+            if (_projectiles.Any())
+            {
+                projectile = _projectiles[0];
```
Comment (ega-ega) [inline Assets/Scripts/Gameplay/Projectiles/ProjectilePool.cs]: ?

File: Assets/Scripts/Gameplay/Projectiles/ProjectilePool.cs [hunk 3]
```diff
@@ -0,0 +1,38 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+using UnityEngine;
+using Zenject;
+
+namespace Gameplay.Projectiles
+{
+    public class ProjectilePool : MonoBehaviour
```
Comment (ega-ega) [inline Assets/Scripts/Gameplay/Projectiles/ProjectilePool.cs]: Why need Monobehaviour?

**Review by ega-ega — CHANGES_REQUESTED** [review]

File: Assets/Scripts/Gameplay/Projectiles/ProjectilePool.cs (hunk 3 above)
Comment (Kjusok) [reply Assets/Scripts/Gameplay/Projectiles/ProjectilePool.cs]: Done

File: Assets/Scripts/Gameplay/Projectiles/ProjectilePool.cs (hunk 1 above)
Comment (Kjusok) [reply Assets/Scripts/Gameplay/Projectiles/ProjectilePool.cs]: Removed

**Review by ega-ega — APPROVED** [review]

**Pull request merged**